test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1806,8 +1805,8 @@ protected TestCluster buildTestCluster(Scope scope, long seed) throws IOExceptio
             ArrayList<Class<? extends Plugin>> mocks = new ArrayList<>(mockPlugins);
             // add both mock plugins - local and tcp if they are not there
             // we do this in case somebody overrides getMockPlugins and misses to call super
-            if (mockPlugins.contains(MockTcpTransportPlugin.class) == false) {
-                mocks.add(MockTcpTransportPlugin.class);
+            if (mockPlugins.contains(randomTestTransportPlugin()) == false) {
+                mocks.add(randomTestTransportPlugin());
```

Comment:
I think that the transport key should align. As the transport that we are using is static, it will be the same for the whole test.

I think we are just added it here in case there was some issue with overriding a method.